controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -112,33 +98,26 @@ protected void startUp() {
         long traceId = LoggerHelpers.traceEnterWithContext(log, this.objectId, "startUp");
         log.info("Initiating controller service startUp");
         log.info("Event processors enabled = {}", serviceConfig.getEventProcessorConfig().isPresent());
-        log.info("Cluster listener enabled = {}", serviceConfig.getControllerClusterListenerConfig().isPresent());
+        log.info("Cluster listener enabled = {}", serviceConfig.isControllerClusterListenerEnabled());
         log.info("    Host monitor enabled = {}", serviceConfig.getHostMonitorConfig().isHostMonitorEnabled());
         log.info("     gRPC server enabled = {}", serviceConfig.getGRPCServerConfig().isPresent());
         log.info("     REST server enabled = {}", serviceConfig.getRestServerConfig().isPresent());
 
+        final StreamMetadataStore streamStore;
+        final TaskMetadataStore taskMetadataStore;
+        final HostControllerStore hostStore;
+        final CheckpointStore checkpointStore;
+
         try {
             //Initialize the executor service.
-            controllerServiceExecutor = Executors.newScheduledThreadPool(serviceConfig.getServiceThreadPoolSize(),
-                    new ThreadFactoryBuilder().setNameFormat("servicepool-%d").build());
-
-            taskExecutor = Executors.newScheduledThreadPool(serviceConfig.getTaskThreadPoolSize(),
-                    new ThreadFactoryBuilder().setNameFormat("taskpool-%d").build());
-
-            storeExecutor = Executors.newScheduledThreadPool(serviceConfig.getStoreThreadPoolSize(),
-                    new ThreadFactoryBuilder().setNameFormat("storepool-%d").build());
-
-            requestExecutor = Executors.newScheduledThreadPool(serviceConfig.getRequestHandlerThreadPoolSize(),
-                    new ThreadFactoryBuilder().setNameFormat("requestpool-%d").build());
-
-            eventProcExecutor = Executors.newScheduledThreadPool(serviceConfig.getEventProcThreadPoolSize(),
-                    new ThreadFactoryBuilder().setNameFormat("eventprocpool-%d").build());
+            controllerExecutor = Executors.newScheduledThreadPool(serviceConfig.getThreadPoolSize(),
```

Comment:
Agree, but I also believe we do not make any blocking requests anywhere in the controller code base except in (1) EventProcessors where we block until receiving an event, but it happens on its independent thread, and (2) We block for all component services to startup and shutdown in ControllerServiceStarter's startup and shutdown methods, respectively, but again that happens on its own independent thread.